### PR TITLE
Nginx 1.12.1

### PR DESCRIPTION
**Pull request by max-lobur**
* Nginx update to 1.12.1
* Configs templated from ppa nginx rpm
* proxy_protocol logging enabled

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -43,17 +53,20 @@
 - name: activate catch all
   file: src={{ ssl_router_nginx_config_dir }}verygoodproxy.conf dest={{ ssl_router_nginx_enabled_dir }}verygoodproxy.conf state=link
 
-- name: render config
-  template: src=templates/nginx.conf.j2 dest={{ ssl_router_nginx_config_dir }}{{ item | basename }}.config mode=0400
+- name: add site configs
+  template: src=templates/site_nginx.conf.j2 dest={{ ssl_router_nginx_config_dir }}{{ item | basename }}.config mode=0400
```
Comment (mjallday) [inline tasks/main.yml]: please call this `vault_tenant.conf.j2`. i think that's clearer.

File: templates/nginx.conf.j2 [hunk 2]
```diff
@@ -1,28 +1,64 @@
...
-      proxy_set_header        X-Real-IP $remote_addr;
-      proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
-      proxy_set_header        X-Forwarded-Proto $scheme;
-      resolver                {{ ansible_dns.nameservers | join (',') }};
-      set                     $vault                                          https://{{ item.split('/')[-2] }};
-      proxy_pass              $vault;
-      proxy_read_timeout      90;
-    }
+user www-data;
+worker_processes auto;
```
Comment (mjallday) [inline templates/nginx.conf.j2]: http://nginx.org/en/docs/ngx_core_module.html#worker_processes

👍

File: templates/nginx.conf.j2 [hunk 3]
```diff
@@ -1,28 +1,64 @@
...
+pid /run/nginx.pid;
+include /etc/nginx/modules-enabled/*.conf;
+
+events {
+	worker_connections 768;
```
Comment (mjallday) [inline templates/nginx.conf.j2]: http://nginx.org/en/docs/ngx_core_module.html#worker_connections

how do we get notified if this limit is exceeded?

File: templates/nginx.conf.j2 [hunk 4]
```diff
@@ -1,28 +1,64 @@
...
+	# server_name_in_redirect off;
+
+	include /etc/nginx/mime.types;
+	default_type application/octet-stream;
+
+	##
+	# SSL Settings
+	##
+
+	ssl_protocols TLSv1 TLSv1.1 TLSv1.2; # Dropping SSLv3, ref: POODLE
```
Comment (mjallday) [inline templates/nginx.conf.j2]: @gjyoung1974 please take a look. we need to decide if we're going to allow insecure protocols or not.

Comment (gjyoung1974) [reply templates/nginx.conf.j2]: We are required to move to TLS 1.1 or better by  30 June 2018 per PCI

We are with v1 until then but shot start planning for it. We can't simply turn it off. we have to study our logs and see which of our customers will be impacted, we will need to hold their hands to help them upgrade. 

For example "Late" TLS v1.1 or better isn't available in Python27 unless you install backports and make some not so obvious code changes. 

It's ok for now but we should backlog or roadmap getting TLS upgraded

Comment (gjyoung1974) [reply templates/nginx.conf.j2]: We also need figgure out what to do with the setting:

ssl_ciphers         HIGH:!aNULL:!MD5;

It's fine for now but in June 30 next year we will need to have tuned it up by then

**Pull request merged**